File: 2020/python/day5/answer.py
import math

def getPosition(line):
  row = [0, 127]
  col = [0, 7]

  for letter in line[0:7]:
    if letter == 'F':
      row[1] = math.floor(sum(row) / 2)
    elif letter == 'B':
      row[0] = math.ceil(sum(row) / 2)

  for letter in line[7:10]:
    if letter == 'L':
      col[1] = math.floor(sum(col) / 2)
    elif letter == 'R':
      col[0] = math.ceil(sum(col) / 2)

  return [row[0], col[0]]

def getID(position):
  return position[0] * 8 + position[1]

def findMissingID(ids):
  current = ids[0]
  for id in ids:
    if current != id:
      break
    current += 1
  return current

with open('input') as f:
  input = f.read().split('\n')
  ids = [getID(getPosition(line)) for line in input ]
  ids.sort()
  
  print(f'Part 1: {ids[-1]}')
  print(f'Part 2: {findMissingID(ids)}')

File: 2020/javascript/day5/answer.js
const fs = require("fs")

const input = fs.readFileSync("input", { encoding: "utf8", flag: "r" }).split('\n')

function getPosition(data) {
  const row = data.substring(0, 7).split('').reduce((pos, letter) => letter == 'F' ? [pos[0], Math.floor((pos[0] + pos[1]) / 2)] : [Math.ceil((pos[0] + pos[1]) / 2),pos[1]], [0, 127])[0]
  const col = data.substring(7).split('').reduce((pos, letter) => letter == 'L' ? [pos[0], Math.floor((pos[0] + pos[1]) / 2)] : [Math.ceil((pos[0] + pos[1]) / 2),pos[1]], [0, 7])[0]
  
  return [row, col]
}

const getID = position => position[0] * 8 + position[1]

const sortedIDs = input.map(getPosition).map(getID).sort((a, b) => a - b)
const missingID = sortedIDs.filter((val, index) => sortedIDs[index - 1] && val - sortedIDs[index - 1] > 1 ? true : false ) - 1

console.log(`Part 1: ${sortedIDs[sortedIDs.length - 1]}`)
console.log(`Part 2: ${missingID}`)